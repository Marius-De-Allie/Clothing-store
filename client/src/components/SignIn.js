import React, { useState } from 'react';
import FormInput from './FormInput';
import CustomButton from './CustomButton';
import { auth, signInWithGoogle } from '../firebase/firebase.utils';
// styles.
import '../styles/signIn.scss';


const SignIn = () => {

    // Component state.
    const [credentials, setCredentials] = useState({email: '', password: ''});

    const { email, password } = credentials;

    const handleChange = evt => {
        const { value, name } = evt.target;
        setCredentials({...credentials, [name]: value})
    };

    const handleSubmit = async evt => {
        evt.preventDefault();
        
        try {
            await auth.signInWithEmailAndPassword(email, password);
            setCredentials({...credentials, email: '', password: ''})
        } catch(error) {
            console.log(error);
        }
        // reset both input fields to an empty string
        setCredentials({...credentials, email: '', password: ''})
    };

    return (
        <div className="sign-in">
            <h2>I already have an account</h2>
            <span>Sign in with your email and password</span>
            <form onSubmit={handleSubmit}>
                <FormInput
                    id="email"
                    type="email" 
                    value={email} 
                    name="email"
                    required
                    handleChange={handleChange}
                    label="email"
                />
                <FormInput 
                    id="password"
                    type="password" 
                    value={password} 
                    name="password"
                    required
                    handleChange={handleChange}
                    label="password"
                />
                <div className="buttons">
                    <CustomButton type="submit"> Sign In </CustomButton>
                    <CustomButton onClick={signInWithGoogle} isGoogleSignIn>Sign In with Google</CustomButton>
                </div>
            </form>
        </div>
    )
};

export default SignIn;