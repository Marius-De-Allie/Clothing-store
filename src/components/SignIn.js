import React from 'react';
import FormInput from './FormInput';
// styles.
import '../styles/signIn.scss';

class SignIn extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            email: '',
            password: ''
        }
    }

    handleChange = evt => {
        const { value, name } = evt.target;
        this.setState(() => ({[name]: value}));
    };

    handleSubmit = evt => {
        evt.preventDefault();
        // reset both input fields to an empty string
        this.setState(() => ({
            email: '',
            password: ''
        }))
    };


    render() {
        return (
            <div className="sign-in">
                <h2>I already have an account</h2>
                <span>Sign in with your email and password</span>
                <form onSubmit={this.handleSubmit}>
                    <FormInput
                        id="email"
                        type="email" 
                        value={this.state.email} 
                        name="email"
                        required
                        handleChange={this.handleChange}
                        label="email"
                    />
                    <FormInput 
                        id="password"
                        type="password" 
                        value={this.state.password} 
                        name="password"
                        required
                        handleChange={this.handleChange}
                        label="password"
                    />
                    <input type="submit" value="submit-form" disable={this.state.email === '' || this.state.password === ''} />
                </form>
            </div>
        )
    }
};

export default SignIn;