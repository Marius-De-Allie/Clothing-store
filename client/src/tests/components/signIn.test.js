import React from 'react';
import { shallow, mount } from 'enzyme';
import SignIn from '../../components/SignIn';

describe('SignIn component', () => {
    it('SignIn component renders properly', () => {
        const wrapper = shallow(<SignIn />);
        expect(wrapper).toMatchSnapshot();
    });
    // it('SignIn component state update correctly', () => {
    //     const wrapper = shallow(<SignIn />);
    //     wrapper.find('#email').simulate('change', { target: { name: 'email', value: 'test@gmail.com' } });
    //     wrapper.find('#password').simulate('change', { target: { name: 'password', value: 'password1' } });
    //     expect(wrapper.state()).toEqual({credentials: {
    //         email: 'test@gmail.com',
    //         password: 'password1'
    //     }});
    // });
    // it('SignIn component onSubmit handler gets called when form is submitted', () => {
    //     const mockonSubmitHandler = jest.fn().mockReturnValue('done');
    //     const wrapper = shallow(<SignIn onSubmit={mockonSubmitHandler} />);
    //     wrapper.find('form').simulate('submit');
    //     expect(mockonSubmitHandler.mock.calls.length).toBe(1);
    // });

    
});
