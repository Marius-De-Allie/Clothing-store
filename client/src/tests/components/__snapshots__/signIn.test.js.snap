// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn component SignIn component renders properly 1`] = `ShallowWrapper {}`;
